### PR TITLE
UTOPIA-859: [Stale data passed error] call piaStateChangeHandler using useEffect for all tabs

### PR DESCRIPTION
**Pull request by Bottle7**
# Description

This PR includes the following proposed change(s):

- Only touches tabs that were missing this implementation
- Currently, storingPersonalInformation tab isn't 100%

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Go through PIA tabs and check for correct behaviour and correct data, i.e.: Autosave is green and the API responds with the correct data.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [x] New and existing unit tests pass locally with my changes
- [x] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/components/public/PIAFormTabs/accuracyRetention/index.tsx [hunk 1]
```diff
@@ -66,12 +73,22 @@ export const AccuracyCorrectionAndRetention = () => {
...
 
+  useEffect(() => {
+    if (!deepEqual(initialFormState, accuracyCorrectionAndRetentionForm)) {
+      piaStateChangeHandler(
+        accuracyCorrectionAndRetentionForm,
+        'accuracyCorrectionAndRetention',
+      );
+    }
+  }, [
+    pia.accuracyCorrectionAndRetention,
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/accuracyRetention/index.tsx:86]: Guess we can remove this dependency

File: src/frontend/src/components/public/PIAFormTabs/additionalRisks/index.tsx [hunk 2]
```diff
@@ -99,6 +128,17 @@ const PIAAdditionalRisks = () => {
 
   const columns = [{ name: 'Possible risk' }, { name: 'Response' }];
 
+  useEffect(() => {
+    if (!deepEqual(initialFormState, additionalRisksForm)) {
+      piaStateChangeHandler(additionalRisksForm, 'additionalRisks');
+    }
+  }, [
+    pia.additionalRisks,
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/additionalRisks/index.tsx:136]: same for this dep `pia.additionalRisks`

You may result in a conflict to Bing's PR.

File: src/frontend/src/components/public/PIAFormTabs/collectionUseAndDisclosure/index.tsx [hunk 3]
```diff
@@ -161,6 +199,21 @@ const PIACollectionUseAndDisclosure = () => {
...
+
+  useEffect(() => {
+    if (!deepEqual(initialFormState, collectionUseAndDisclosureForm)) {
+      piaStateChangeHandler(
+        collectionUseAndDisclosureForm,
+        'collectionUseAndDisclosure',
+      );
+    }
+  }, [
+    pia.collectionUseAndDisclosure,
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/collectionUseAndDisclosure/index.tsx:211]: same for `pia.collectionUseAndDisclosure`

**Review by kushal-arora-fw — COMMENTED** [review]
Looks goods. Thanks for picking this up, Brandon! It was on my plate for so long.

Added some minor comments.

**Review by binggaofw — DISMISSED** [review]

File: src/frontend/src/components/public/PIAFormTabs/collectionUseAndDisclosure/index.tsx (hunk 3 above)
Comment (kushal-arora-fw) [reply src/frontend/src/components/public/PIAFormTabs/collectionUseAndDisclosure/index.tsx:204]: missed this

Comment (Bottle7) [reply src/frontend/src/components/public/PIAFormTabs/collectionUseAndDisclosure/index.tsx]: There were duplicate useEffects for two tabs which I didn't notice the first time around.

**Comment by kushal-arora-fw** [comment]
Docker build is failing @Bottle7

**Review by antsand — APPROVED** [review]
Looks good. Learnt a new react function usememo

**Review by kushal-arora-fw — APPROVED** [review]

**Pull request merged**